ConnectionExceptions thrown from Bugzilla.connectTo()

Tmarks opened the pull request:
Added the ConnectionException class from the old j2bugzilla code, and made all overloads of Bugzilla.connectTo() throw this exception, just as the BugzillaConnector did.

Tmarks committed:
Merge pull request #1 from TomRK1089/master

get latest

Tmarks committed:
Added the ConnectionException. It's taken straight from the old J2Bugzilla implementation. It belonged here because the BugzillaConnector throws it from its connectTo() methods, but the Bugzilla class did not.
This was also changed; Bugzilla's connectTo() methods now throw ConnectionException in their signatures.

Thorn1089 commented on src/main/java/com/j2bugzilla/api/ConnectionException.java:
```diff
@@ -0,0 +1,45 @@
+ * @author Tom
+ *
+ */
+public class ConnectionException extends Exception {
```
When I first wrote the alpha for J2Bugzilla, I was solidly in the checked exception camp. Now I'm wondering if this ought to extend RuntimeException instead. On the one hand, it's nice to provide callers something specific to catch instead of plain old Exception; on the other hand, it depends if you really consider a low-level network problem of the type encountered by the connect methods to really be recoverable. I feel like RuntimeExceptions strike a slightly better balance of "Catch it if you want to be robust, let it fly if you want things to blow up loudly."

Thorn1089 commented on src/main/java/com/j2bugzilla/api/ConnectionException.java:
```diff
@@ -0,0 +1,45 @@
+	 * @param message A custom error message describing the issue
+	 * @param cause The root cause of the exception
+	 */
+	public ConnectionException(String message, Throwable cause) {
```
I would probably override the 4 most common exception constructors.

Tmarks committed:
SearchBy.java:
 - *Finally* changed getName()'s visibility to public...
 - Fixed a comment on the constructor to say SearchBy instead of SearchLimiter. Must've been an old name.
pom.xml:
 - Added dependency for xmlrpc-client.
Bugzilla.java:
 - Removed throws declarations of ConnectionException from all methods, since...
ConnectionException.java:
 - ... now extends RuntimeException.
Bugzilla.java:
